test/CMakeLists.txt

```diff
@@ -17,28 +17,46 @@ else ()
     set(VARIANT --noJit)
 endif()
 
+# Target to run all tests
+add_custom_target(check)
+
 if (CMAKE_BUILD_TYPE STREQUAL Release)
-    set(TEST_SUITE echo "Running Hello World test" && ${CMAKE_BINARY_DIR}/ch ${CMAKE_CURRENT_SOURCE_DIR}/Basics/hello.js)
+    add_custom_target(smoke-check
+        COMMAND echo "Running Hello World test"
+        COMMAND ${CMAKE_BINARY_DIR}/ch ${CMAKE_CURRENT_SOURCE_DIR}/Basics/hello.js
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        USES_TERMINAL
+        DEPENDS ch
+        )
+    add_dependencies(check smoke-check)
 else()
-    set(BYTE_CODE_CHECK ${CMAKE_CURRENT_SOURCE_DIR}/../tools/regenByteCode.py ${VARIANT} --verify --binary=${CMAKE_BINARY_DIR}/ch)
-    set(TEST_SUITE ${CMAKE_CURRENT_SOURCE_DIR}/runtests.py ${TEST_BUILD_TYPE} ${TEST_ICU} ${TEST_VARIANT} --binary=${CMAKE_BINARY_DIR}/ch)
+    add_custom_target(bytecode-check
+        COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/regenByteCode.py ${VARIANT} --verify --binary=${CMAKE_BINARY_DIR}/ch
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        USES_TERMINAL
+        DEPENDS ch
+        )
+    add_custom_target(regression-check
+        COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/runtests.py ${TEST_BUILD_TYPE} ${TEST_ICU} ${TEST_VARIANT} --binary=${CMAKE_BINARY_DIR}/ch
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        USES_TERMINAL
+        DEPENDS ch
+        )
+    add_dependencies(check bytecode-check regression-check)
 endif()
 
 if (NOT NO_ICU AND NOT EMBED_ICU)
-    set(NATIVE_TEST cd ${CMAKE_CURRENT_SOURCE_DIR}/native-tests && ./test_native.sh ${CMAKE_BINARY_DIR}/ch ${CMAKE_BUILD_TYPE} ${CMAKE_C_COMPILER} ${CMAKE_CXX_COMPILER} ${CMAKE_SOURCE_DIR})
+    add_custom_target(native-check
+        COMMAND ./test_native.sh ${CMAKE_BINARY_DIR}/ch ${CMAKE_BUILD_TYPE} ${CMAKE_C_COMPILER} ${CMAKE_CXX_COMPILER} ${CMAKE_SOURCE_DIR}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/native-tests
+        USES_TERMINAL
+        DEPENDS ch
+        )
+    add_dependencies(check native-check)
 else()
-    set(NATIVE_TEST echo "Native tests NOT RUN, they require ICU to be installed not embedded. NO_ICU = ${NO_ICU} EMBED_ICU = ${EMBED_ICU}")
+    message("Native tests are DISABLED, as they require ICU to be installed not embedded. NO_ICU = ${NO_ICU} EMBED_ICU = ${EMBED_ICU}")
```

Comment:
nit: does this message show anywhere in the CI, I can't see it? Not the most important but with the old setup the message about not running the native tests was printed when running the rest of the tests, good if comparing test logs between builds and wondering why the native tests are missing